File: development/sphinx-console-web/src/main/webapp/sphinx-console/app/model/DistributedCollectionWrapper.js
Ext.define('sphinx-console.model.DistributedCollectionWrapper', {
    extend: 'Ext.data.Model',
    autoLoad: true,
    autoSync: true,
    hasOne: [
                {model: 'sphinx-console.model.Collection', getterName: 'getCollection', setterName: 'setCollection', associationKey: 'collection', associatedName: 'collection' },
                {model: 'sphinx-console.model.SearchConfigurationPort', getterName: 'getSearchConfigurationPort', setterName: 'setSearchConfigurationPort', associationKey: 'searchConfigurationPort', associatedName: 'searchConfigurationPort' },
                {model: 'sphinx-console.model.Configuration', getterName: 'getSearchConfiguration', setterName: 'setSearchConfiguration', associationKey: 'searchConfiguration', associatedName: 'searchConfiguration', name: 'searchConfiguration', instanceName: 'searchConfiguration', foreingKey: 'id'},
                {model: 'sphinx-console.model.Server', getterName: 'getSearchServer', setterName: 'setSearchServer', associationKey: 'searchServer', associatedName: 'searchServer'}
            ],
    hasMany:[
        {model: 'sphinx-console.model.SimpleCollectionWrapper', name: 'nodes', associationKey: 'nodes'}
    ]        
});

File: development/sphinx-console-coordinator/src/main/java/ru/skuptsov/sphinx/console/coordinator/model/DistributedCollectionWrapper.java
package ru.skuptsov.sphinx.console.coordinator.model;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;

public class DistributedCollectionWrapper implements Serializable {
    private static final long serialVersionUID = 190647L;
    
    private List<SimpleCollectionWrapper> nodes = new ArrayList<SimpleCollectionWrapper>();
    
    @NotNull
    @Valid
    private SearchConfigurationPortWrapper searchConfigurationPort;
    @NotNull
    @Valid
	private Configuration searchConfiguration;
    @NotNull
    @Valid
    private Server searchServer;
    
    @NotNull
    @Valid
    private Collection collection;
    

	public List<SimpleCollectionWrapper> getNodes() {
		return nodes;
	}

	public void setNodes(List<SimpleCollectionWrapper> nodes) {
		this.nodes = nodes;
	}

	public SearchConfigurationPortWrapper getSearchConfigurationPort() {
		return searchConfigurationPort;
	}

	public void setSearchConfigurationPort(
			SearchConfigurationPortWrapper searchConfigurationPort) {
		this.searchConfigurationPort = searchConfigurationPort;
	}

	

	public Configuration getSearchConfiguration() {
		return searchConfiguration;
	}

	public void setSearchConfiguration(Configuration searchConfiguration) {
		this.searchConfiguration = searchConfiguration;
	}

	

	public Server getSearchServer() {
		return searchServer;
	}

	public void setSearchServer(Server searchServer) {
		this.searchServer = searchServer;
	}

	public Collection getCollection() {
		return collection;
	}

	public void setCollection(Collection collection) {
		this.collection = collection;
	}

	
    
    
    
}
